toplevel/coqloop.ml

```diff
@@ -337,11 +336,15 @@ let loop_flush_all () =
 
 let rec loop () =
   Sys.catch_break true;
-  if !Flags.print_emacs then Vernacentries.qed_display_script := false;
-  Flags.coqtop_ui := true;
   try
     reset_input_buffer stdin top_buffer;
-    while true do do_vernac(); loop_flush_all () done
+    (* Tail-rec is essential *)
+    let rec loop sid =
```

Comment:
This inner loop should not have the name of the outer loop.

My personal taste leans toward
```
let cur := ref (Stm.get_current_state()) in
while true do cur := do_vernac !cur; flush(); done
```
I mean, if it has to be a loop, let it be a real loop ;-) ... I know, I'm how a Haskell guy...